salt/template.py

```diff
@@ -88,8 +88,9 @@ def compile_template(template,
         if not isinstance(input_data, (dict, OrderedDict)):
             try:
                 input_data.seek(0)
-            except Exception as exp:
-                log.error('error while compiling template \'{0}\': {1}'.format(template, exp))
+            except AttributeError:
```

Comment:
@cachedout Since the reasoning for the general exception handling is lost, do you still require a code change to accept this PR?